agent/handlers/task_server_setup_sudo_integ_test.go

```diff
@@ -0,0 +1,427 @@
+		body2         interface{}
+		responseCode1 int
+		responseCode2 int
+	}{
+		{
+			name:          "network latency same type",
+			faultType1:    faulttype.LatencyFaultType,
+			faultType2:    faulttype.LatencyFaultType,
+			body1:         happyNetworkLatencyReqBody,
+			body2:         happyNetworkLatencyReqBody,
+			responseCode1: http.StatusOK,
+			responseCode2: http.StatusConflict,
+		},
+		{
+			name:          "network packet loss same type",
+			faultType1:    faulttype.PacketLossFaultType,
+			faultType2:    faulttype.PacketLossFaultType,
+			body1:         happyNetworkPacketLossReqBody,
+			body2:         happyNetworkPacketLossReqBody,
+			responseCode1: http.StatusOK,
+			responseCode2: http.StatusConflict,
+		},
+		{
+			name:          "network latency and packet loss different type",
+			faultType1:    faulttype.LatencyFaultType,
+			faultType2:    faulttype.PacketLossFaultType,
+			body1:         happyNetworkLatencyReqBody,
+			body2:         happyNetworkPacketLossReqBody,
+			responseCode1: http.StatusOK,
+			responseCode2: http.StatusConflict,
+		},
+		{
+			name:          "network black hole port same type",
+			faultType1:    faulttype.BlackHolePortFaultType,
+			faultType2:    faulttype.BlackHolePortFaultType,
+			body1:         happyBlackHolePortReqBody,
+			body2:         happyBlackHolePortReqBody,
+			responseCode1: http.StatusOK,
+			responseCode2: http.StatusOK,
+		},
+		{
+			name:          "network black hole port and latency different type",
+			faultType1:    faulttype.BlackHolePortFaultType,
+			faultType2:    faulttype.LatencyFaultType,
+			body1:         happyBlackHolePortReqBody,
+			body2:         happyNetworkLatencyReqBody,
+			responseCode1: http.StatusOK,
+			responseCode2: http.StatusOK,
+		},
+		{
+			name:          "network black hole port and packet loss different type",
+			faultType1:    faulttype.BlackHolePortFaultType,
+			faultType2:    faulttype.PacketLossFaultType,
+			body1:         happyBlackHolePortReqBody,
+			body2:         happyNetworkPacketLossReqBody,
+			responseCode1: http.StatusOK,
+			responseCode2: http.StatusOK,
+		},
+	}
+
+	skipForUnsupportedTc(t)
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			// Starting a server which will include the network fault handlers
+			server, port := startServer(t)
+
+			client := &http.Client{}
+
+			// Making request 1
+			bodyBytes, _ := json.Marshal(tc.body1)
+			bodyReader := bytes.NewReader(bodyBytes)
+			req1, err := http.NewRequest(http.MethodPost, getURL(port, fmt.Sprintf(startEndpoint, endpointId, tc.faultType1)), bodyReader)
+			require.NoError(t, err)
+
+			res1, err := client.Do(req1)
+			require.NoError(t, err)
+
+			// Making request 2
+			var resp2Code int
+			// Adding a retry mechanism since there's a rate limit of only one request per 5 seconds for each network fault endpoints
+			// This is in the case of making the same requests type back to back
+			for i := 0; i < requestMaxRetries; i++ {
```

Comment:
nit: Why is sleeping slightly longer than 5 seconds not enough and we need retries?

Reply:
there are cases where sleeping isn't needed. thought having a retry would be better

Reply:
sleeping for 6 seconds